src/traces/parcoords/axisbrush.js

```diff
@@ -0,0 +1,520 @@
+        ? ( // if we're South of the 1st interval, there's no previous interval
+            isNaN(previousInterval)
+                ? nextInterval
+                : ( // if we're North of the last interval, there's no next interval
+                    isNaN(nextInterval)
+                        ? previousInterval
+                        : ( // if we have both previous and subsequent intervals, which one is closer?
+                            y - pixIntervals[previousInterval][1] < pixIntervals[nextInterval][0] - y
+                                ? previousInterval
+                                : nextInterval)))
+        : hoveredInterval; // if we're hovering over an interval, that's trivially the closest interval
+    var fPix = pixIntervals[closestInterval];
+
+    return {
+        interval: isNaN(closestInterval) ? null : intervals[closestInterval], // activated interval in domain terms
+        intervalPix: isNaN(closestInterval) ? null : pixIntervals[closestInterval], // activated interval in pixel terms
+        n: north(fPix, y), // do we hover over the northern resize hotspot
+        s: south(fPix, y), // do we hover over the northern resize hotspot
+        m: middle(fPix, y) // or over the bar section itself?
+    };
+}
+
+function attachDragBehavior(selection) {
+    // There's some fiddling with pointer cursor styling so that the cursor preserves its shape while dragging a brush
+    // even if the cursor strays from the interacting bar, which is bound to happen as bars are thin and the user
+    // will inevitably leave the hotspot strip. In this regard, it does something similar to what the D3 brush would do.
+    selection
+        .on('mousemove', function(d) {
+            d3.event.preventDefault();
+            var b = d.brush;
+            if(d.parent.inBrushDrag) {
+                return;
+            }
+            var y = d.unitScaleInOrder(d.unitScale.invert(d3.mouse(this)[1]));
+            var interval = getInterval(b, d.unitScaleInOrder, y);
+            d3.select(document.body)
+                .style('cursor', interval.n ? 'n-resize' : interval.s ? 's-resize' : !interval.m ? 'crosshair' : filterActive(b) ? 'ns-resize' : 'crosshair');
+        })
+        .on('mouseleave', function(d) {
+            if(d.parent.inBrushDrag) {
+                return;
+            }
+            clearCursor();
+        })
+        .call(d3.behavior.drag()
+            .on('dragstart', function(d) {
+                var e = d3.event;
+                e.sourceEvent.stopPropagation();
+                var y = d.unitScaleInOrder(d.unitScale.invert(d3.mouse(this)[1]));
+                var unitLocation = d.unitScaleInOrder.invert(y);
+                var b = d.brush;
+                var intData = getInterval(b, d.unitScaleInOrder, y);
+                var unitRange = intData.interval;
+                var pixelRange = unitRange.map(d.unitScaleInOrder);
+                var s = b.svgBrush2;
```

Comment:
upcoming commit will remove the `2` post-fix (we used to have both D3 and our own brush for manual regression testing but not anymore)